src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -114,22 +117,27 @@ private static Type MaxKind(Type a, Type b)
         /// <param name="guessRows">number of rows used to guess types</param>
         /// <param name="addIndexColumn">add one column with the row index</param>
         /// <returns>DataFrame</returns>
-        public static DataFrame ReadStream(Func<StreamReader> createStream,
+        public static DataFrame LoadStream(Stream csvStream,
                                 char separator = ',', bool header = true,
                                 string[] columnNames = null, Type[] dataTypes = null,
                                 long numberOfRowsToRead = -1, int guessRows = 10, bool addIndexColumn = false)
         {
+            if (!csvStream.CanSeek)
+                throw new ArgumentException(Strings.NonSeekableStream, nameof(csvStream));
+
             var linesForGuessType = new List<string[]>();
             long rowline = 0;
             int numberOfColumns = 0;
 
             if (header == true && numberOfRowsToRead != -1)
                 numberOfRowsToRead++;
 
+            List<DataFrameColumn> columns;
+            long streamStart = csvStream.Position;
             // First pass: schema and number of rows.
-            using (var st = createStream())
+            using (var streamReader = new StreamReader(csvStream))
```

Comment:
I believe the `StreamReader` will dispose the underlying stream when the `StreamReader` is disposed.

Is it expected that calling `DataFrame.LoadStream(someStream)` is going to dispose the `someStream`?